alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt

```diff
@@ -525,12 +547,321 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
+            )
+        } catch (e: ResponseException) {
+            assertEquals("Unexpected status", RestStatus.FORBIDDEN, e.response.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE, ALERTING_NO_ACCESS_ROLE)
+        }
+    }
+
+    fun `test delete email accounts with an user with delete email account role`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            TEST_HR_BACKEND_ROLE,
+            getClusterPermissionsFromCustomRole(ALERTING_DELETE_EMAIL_ACCOUNT_ACCESS)
+        )
+
+        val emailAccount = createRandomEmailAccountWithGivenName(true, randomAlphaOfLength(5))
+
+        try {
+            val deleteEmailAccountsResponse = userClient?.makeRequest(
+                "DELETE",
+                "${AlertingPlugin.EMAIL_ACCOUNT_BASE_URI}/${emailAccount.id}",
+                null,
+                BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+            )
+
+            assertEquals("Get Email failed", RestStatus.OK, deleteEmailAccountsResponse?.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE, ALERTING_DELETE_EMAIL_ACCOUNT_ACCESS)
+        }
+    }
+
+    fun `test delete email accounts with an user without delete email account role`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            TEST_HR_BACKEND_ROLE,
+            getClusterPermissionsFromCustomRole(ALERTING_NO_ACCESS_ROLE)
+        )
+
+        val emailAccount = createRandomEmailAccountWithGivenName(true, randomAlphaOfLength(5))
+
+        try {
+            userClient?.makeRequest(
+                "DELETE",
+                "${AlertingPlugin.EMAIL_ACCOUNT_BASE_URI}/${emailAccount.id}",
+                null,
+                BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+            )
+        } catch (e: ResponseException) {
+            assertEquals("Unexpected status", RestStatus.FORBIDDEN, e.response.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE, ALERTING_NO_ACCESS_ROLE)
+        }
+    }
+
+    // Email groups related tests.
+
+    fun `test index and search email groups with an user with search and index email group role`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            TEST_HR_BACKEND_ROLE,
+            getClusterPermissionsFromCustomRole(ALERTING_SEARCH_AND_INDEX_EMAIL_GROUP_ACCESS)
+        )
+
+        try {
+            val emailGroup = createRandomEmailGroupWithGivenName(true, randomAlphaOfLength(5))
+            assertEquals("Index Email Group failed", false, emailGroup.id.isEmpty())
+            val searchEmailGroupsResponse = searchEmailGroups(client())
+            assertEquals("Search Email Group failed", RestStatus.OK, searchEmailGroupsResponse.restStatus())
```

Comment:
No data is being index since no index call is made. Also check the search response to make sure you are getting back the right data. That way you would also know that this test should have failed.

Reply:
moved to a separate file and added the call